src/Pages/Nodes.razor

```diff
@@ -235,9 +282,28 @@
         }
         else
         {
-            arg.Item.UpdateDatetime = DateTimeOffset.Now;
-            arg.Item.CreationDatetime = DateTimeOffset.Now;
-            var addResult = await NodeRepository.AddAsync(arg.Item);
+            Node newNode = arg.Item as Node;
+            (bool, string?) addResult;
+            var restoredNode = await NodeRepository.GetByPubkey(arg.Item.PubKey);
+         
+            if (restoredNode != null)
+            {
+                newNode = restoredNode;
+                newNode.Endpoint = arg.Item.Endpoint;
+                newNode.ChannelAdminMacaroon = arg.Item.ChannelAdminMacaroon;
+                newNode.IsNodeEnabled = true;
+                newNode.Description =  arg.Item.Description;
+                newNode.AutosweepEnabled = arg.Item.AutosweepEnabled;
+                newNode.Name = arg.Item.Name;
+                newNode.ReturningFundsWalletId = arg.Item.ReturningFundsWalletId;
```

Comment:
have you tried adding the node id to the new item and calling update? Maybe it works
```
Node newNode = arg.Item as Node;
newNode.Id = restoredNode.Id
addResult = NodeRepository.Update(newNode);
```

Reply:
Yes, it works.

Reply:
Ok, it could be improved.